### ISSUE TITLE
Add touch support

### ISSUE THREAD

**Issue by smeijer** [leefsmp/Re-Flex#24]
It's not really working on touch screens (my notebook has both, a mouse and a touch screen). Sometimes I'm able to grab the handle, and move it a little. But most often not.

For a mouse user, awesome library! Thanks.

**Comment by leefsmp** [comment]
There is already support for touch screens, this is done by tweaking the `reflex-splitter` style adding `reflex-thin` style. By default this style is activated for mobile platforms using the code in [Browser.js](https://github.com/leefsmp/Re-Flex/blob/master/src/lib/Browser.js#L85). On your machine `isMobile` probably returns false and default style is applied. A simple workaround would be that you manually force `reflex-thin` on the splitter: `<ReflexSplitter className="reflex-thin">`.

Alternatively you can take a look at [reflex-styles.scss](https://github.com/leefsmp/Re-Flex/blob/master/src/lib/reflex-styles.scss) and overwrite the default style to suit your needs or add your styles in your own css class than applied it to the splitter as above using `className`.

**Comment by smeijer** [comment]
`reflex-thin` seems to help, but I guess there is a `preventDefault` missing? When dragging to the right, chrome starts to navigate to the previous page:

![image](https://user-images.githubusercontent.com/1196524/29729123-1fbfc4d6-89db-11e7-8891-70a6192770da.png)

**Comment by leefsmp** [comment]
`preventDefault` is there, check the [ReflexSplitter source](https://github.com/leefsmp/Re-Flex/blob/master/src/lib/ReflexSplitter.jsx). 

I cannot reproduce this issue on my side, works fine in the mobile emulator as well as on iOS and Android... What platform/browser are you running that? And when does the navigation occur, is it on mouse move?

**Comment by smeijer** [comment]
It happens when the `chrome flag` `overscroll-history-navigation` is set to `enabled`. Disabling it fixes the behavior. Although it is an experimental function; I cannot remember that I've enabled it myself.

`chrome://flags/#overscroll-history-navigation`

![image](https://user-images.githubusercontent.com/1196524/29742804-f2f779b6-8a85-11e7-9d18-144b0bac9fa1.png)

Using Chrome `61.0.3163.59` on Windows `10.1703`, both 64 bit.

The navigation occurs when dragging the splitter with the finger on the touch screen. The mouse (touchpad) is behaving as expected. It's because my device has both; a touchpad and a touchscreen.

**Comment by smeijer** [comment]
I'm guessing it has to do with the warning in my console:

```
ReflexSplitter.js? [Intervention] Unable to preventDefault inside passive event listener due to target being treated as passive. See https://www.chromestatus.com/features/5093566007214080
onMouseMove @ ReflexSplitter.js?:134
```

**Comment by smeijer** [comment]
https://github.com/leefsmp/Re-Flex/blob/3e45856e822549bc2c6b2a469af4c6f79fe78c7e/src/lib/ReflexSplitter.jsx#L82:L85

Changing these lines in [`ReflexSplitter:L82:L85`](https://github.com/leefsmp/Re-Flex/blob/3e45856e822549bc2c6b2a469af4c6f79fe78c7e/src/lib/ReflexSplitter.jsx#L82:L85) to set the handler to non-passive, fixes the navigation issue.

```js
this.document.addEventListener(
  'touchmove',
  this.onMouseMove,
  { passive: false } // added to fix the unwanted back / forward navigation
);
```

The warning I spoke of earlier is still thrown by some of the events; but it's no longer causing any issues. I do however think that some of the `preventDefault`s or `stopPropagation` aren't really necessary. Especially since they're already being ignored anyway.

**Comment by leefsmp** [comment]
Awesome! I fixed that in v 2.0.9 and removed `preventDefault stopPropagation` in mouse down event. Thanks.
